crates/oxc_linter/src/rules/eslint/no_import_assign.rs

```diff
@@ -47,14 +49,47 @@ impl Rule for NoImportAssign {
         let symbol_table = ctx.semantic().symbols();
         if symbol_table.get_flag(symbol_id).is_import_binding() {
             for reference in symbol_table.get_resolved_references(symbol_id) {
-                if reference.is_write() {
+                if reference.is_write()
+                    || is_invalid_assign(reference.node_id(), ctx.semantic().nodes())
+                {
                     ctx.diagnostic(NoImportAssignDiagnostic(reference.span()));
                 }
             }
         }
     }
 }
 
+fn is_invalid_assign(current_node_id: AstNodeId, nodes: &AstNodes) -> bool {
+    if let Some((curr, parent)) = nodes
+        .iter_parents(nodes.parent_id(current_node_id).unwrap_or(current_node_id))
+        .tuple_windows::<(&AstNode<'_>, &AstNode<'_>)>()
+        .next()
+    {
+        match (curr.kind(), parent.kind()) {
+            (
+                AstKind::MemberExpression(_),
+                AstKind::SpreadElement(_)
+                | AstKind::ObjectProperty(_)
+                | AstKind::PropertyKey(_)
+                | AstKind::AssignmentTargetWithDefault(_)
+                | AstKind::AssignmentTarget(_)
+                | AstKind::ForOfStatement(_)
+                | AstKind::ForInStatement(_)
+                | AstKind::MemberExpression(_),
+            ) => {
+                return false;
+            }
+            (_, AstKind::MemberExpression(expr)) | (AstKind::MemberExpression(expr), _) => {
+                return expr.static_property_name() != Some("prop");
+            }
```

Comment:
I don't understand why we need to check `static_property_name`. Can you explain it?

Reply:
i did this check to pass these conditions 

![image](https://github.com/oxc-project/oxc/assets/46359732/59404a86-97a3-49ca-93d6-a567b40b5334)

and fails over these

![image](https://github.com/oxc-project/oxc/assets/46359732/b8bad17e-1738-41cc-8e3b-8c2c306554f7)

Reply:
> i did this check to pass these conditions
> 
> ![image](https://private-user-images.githubusercontent.com/46359732/300290500-59404a86-97a3-49ca-93d6-a567b40b5334.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDY0OTQ1MjQsIm5iZiI6MTcwNjQ5NDIyNCwicGF0aCI6Ii80NjM1OTczMi8zMDAyOTA1MDAtNTk0MDRhODYtOTdhMy00OWNhLTkzZDYtYTU2N2I0MGI1MzM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTI5VDAyMTAyNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM5NTcyMjQ5NmM3ODVlNDIxMGQzODRhYmNkYWNmODg0OWZlZWM3Y2RjZWQ5NjI2Njg0MzgyNGJmYzc2MTM2YWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.IoHxPsnBzlMDCSKKUcDToo-kooim9P35oH-nfyFuwQA)
> 

No, You shouldn't check this with `prop` name. We should forbid all property changes. Not just `prop`